src/main/java/io/cdap/plugin/servicenow/source/ServiceNowMultiInputFormat.java

```diff
@@ -0,0 +1,161 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.servicenow.source;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.servicenow.source.apiclient.ServiceNowTableAPIClientImpl;
+import io.cdap.plugin.servicenow.source.apiclient.ServiceNowTableDataResponse;
+import io.cdap.plugin.servicenow.source.util.SchemaBuilder;
+import io.cdap.plugin.servicenow.source.util.ServiceNowColumn;
+import io.cdap.plugin.servicenow.source.util.ServiceNowConstants;
+import io.cdap.plugin.servicenow.source.util.ServiceNowTableInfo;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * ServiceNow input format.
+ */
+public class ServiceNowMultiInputFormat extends InputFormat<NullWritable, StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowInputFormat.class);
+
+  /**
+   * Updates the jobConfig with the ServiceNow table information, which will then be read in getSplit() function.
+   *
+   * @param jobConfig the job configuration
+   * @param conf the database conf
+   * @return Collection of ServiceNowTableInfo containing table and schema.
+   */
+  public static Set<ServiceNowTableInfo> setInput(Configuration jobConfig,
+                                                   ServiceNowMultiSourceConfig conf) {
+    ServiceNowJobConfiguration jobConf = new ServiceNowJobConfiguration(jobConfig);
+    jobConf.setMultiSourcePluginConfiguration(conf);
+
+    // Depending on conf value fetch the list of fields for each table and create schema object
+    // return the schema object for each table as ServiceNowTableInfo
+    Set<ServiceNowTableInfo> tableInfos = fetchTablesInfo(conf);
+
+    jobConf.setTableInfos(tableInfos.stream().collect(Collectors.toList()));
+
+    return tableInfos;
+  }
+
+  public static Set<ServiceNowTableInfo> fetchTablesInfo(ServiceNowMultiSourceConfig conf) {
```

Comment:
why is this method public?

Reply:
Changed the access modifier to `protected` 7661e08